lmfit/model.py

```diff
@@ -1299,28 +1318,49 @@ def _buildmodel(state, funcdefs=None):
         fcn = getattr(lineshapes, fname, None)
         if callable(fcn):
             known_funcs[fname] = fcn
-    if funcdefs is not None:
+    if funcdefs is None:
+        funcdefs = {}
+    else:
         known_funcs.update(funcdefs)
 
     left, right, op = state
     if op is None and right is None:
-        (fname, fcndef, name, prefix, ivars, pnames,
-         phints, nan_policy, opts) = left
-        if not callable(fcndef) and fname in known_funcs:
-            fcndef = known_funcs[fname]
+        if isinstance(left, tuple) and len(left) == 9:
+            (fname, func, name, prefix, ivars, pnames,
+             phints, nan_policy, opts) = left
+        elif isinstance(left, dict) and 'version' in left:
+            if left['version'] == '2':
```

Comment:
this is always "True"

Reply:
@reneeotten Thanks - I'll look at this.  I think the idea was to attempt to be future-proof.